src/software/amazon/ion/IonSystem.java

```diff
@@ -299,9 +311,33 @@ public SymbolTable newSharedSymbolTable(String name,
      * @return a new iterator instance.
      *
      * @throws NullPointerException if <code>ionData</code> is null.
+     *
+     * @deprecated Will cause a memory leak when reading a gzipped byte[].
+     * Use {@link IonSystem#iterate(IonReader)} instead.
      */
+    @Deprecated
     public Iterator<IonValue> iterate(byte[] ionData);
 
+    /**
+     * <p>
+     * Creates an iterator over Ion data.
+     * Values returned by the iterator have no container.
+     * </p>
+     * <p>
+     * The iterator will automatically consume Ion system IDs and local symbol
+     * tables; they will not be returned by the iterator.
+     * </p>
+     * <p>
+     * This method will auto-detect and uncompress GZIPped Ion data.
```

Comment:
remove line, not relevant to this method (but it _is_ relevant to the method that creates the IonReader)